### PR TITLE
Remove leading `$` from shell commands.

### PR DESCRIPTION
**Pull request by EriKWDev**
Please don't make it harder than it has to be to copy commands. I don't understand why every project does this. Especially for multiline commands. Commands are already in a code block.

```bash
$ git clone https://github.com/poacpm/poac.git
$ cd poac
$ cmake -B build -G Ninja -DCMAKE_BUILD_TYPE=Release
$ cd build
$ ninja
$ ninja install
zsh: $: command not found...
zsh: $: command not found...
zsh: $: command not found...
zsh: $: command not found...
zsh: $: command not found...
zsh: $: command not found...
```

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -221,11 +221,11 @@ you:~/hello_world$ poac fmt
...
-> $ # We need to avoid the `build` dir and such dirs ...
-> $ clang-format ./src/*.cpp -i
-> $ clang-format ./include/**/*.hpp -i
-> $ clang-format ./tests/**/*.cpp -i
-> $ ...
+> # We need to avoid the `build` dir and such dirs ...
+> clang-format ./src/*.cpp -i
+> clang-format ./include/**/*.hpp -i
+> clang-format ./tests/**/*.cpp -i
+> ...
```
Comment (ken-matsui) [inline README.md:228]: This is just an example; not intended to be executed.

File: README.md [hunk 2]
```diff
@@ -235,7 +235,8 @@ To customize the format settings, try creating a [`.clang-format`](/.clang-forma
 In case you would find what packages are provided, you can use the `search` command or visit [poac.pm](https://poac.pm).
 
 ```console
-$ poac search func
+poac search func
+
```
Comment (ken-matsui) [inline README.md:239]: This is showing a console output

File: README.md [hunk 3]
```diff
@@ -412,7 +413,7 @@ https://dev.poac.pm
 ### Linter
 
 ```bash
-$ cpplint --quiet --recursive .
+cpplint --quiet --recursive .
 ```
 
 > **Note**:
```
Comment (ken-matsui) [inline README.md:419]: Why not remove `$` from the following command?

**Comment by wx257osn2** [comment]
@ken-matsui IMHO, leading `$` doesn't has to be removed, but you should use `console` (or `bash session` ) highlight instead of `bash` highlight.

**Comment by ken-matsui** [comment]
@wx257osn2 He means copying the snippet using [the button](https://twitter.com/natfriedman/status/1391862005095485440?s=21), so that makes sense to me. Can I have why you prefer the `console` highlight?

Currently, I am using `console` as commands with outputs and `bash` as commands with no outputs.

**Comment by wx257osn2** [comment]
@ken-matsui Because the prompt is not a part of shell script. When you indicate the codes as `bash` script, you should not contains the console prompt `$` to it since it should be executable as bash script without any edit like removing leading `$` .

**Comment by EriKWDev** [comment]
I have never seen this project before and it's just a pet peeve of mine when people add $ or > or whatever in front of commands in the readme/docs.

It makes it annoying anytime a user actually wants to just copy that command that you've so nicely already written for them. Especially for multiline statements.

I don't care if you keep or remove some, but if you have perfectly fine installation instructions, don't ruin them with something so silly. This is true for any project, not just this one.

For example, the entire rust book's commands all start with $ for no reason.

File: README.md [hunk 4]
```diff
@@ -221,11 +221,11 @@ you:~/hello_world$ poac fmt
 > This command automatically detects what files we need to format to avoid bothering commands like:
 >
 > ```bash
```
Comment (ken-matsui) [inline README.md:223]: Please change this to `console`

File: README.md (hunk 1 above)
Comment (ken-matsui) [reply README.md:228]: Please revert this change.

File: README.md (hunk 2 above)
Comment (ken-matsui) [reply README.md:239]: Please revert this

File: README.md (hunk 3 above)
Comment (ken-matsui) [reply README.md:419]: Please remove `$` from `$ poac lint` in the following.

**Review by ken-matsui — COMMENTED** [review]
Thanks y'all, that makes sense. I've added comments so please update this PR as them. After that, LGTM.

**Comment by wx257osn2** [comment]
@EriKWDev  Adding prompt is not necessarily fatuous thing. For example:

```console
host:/path/to$ docker run -it ubuntu:22.04 /bin/bash
docker:/# echo foo
foo
```

represents the working directory, environment, and user permissions in easy-to-understand way.
It also indicates that `foo` at 3rd line is output of `echo foo` at 2nd line (@ken-matsui uses `console` highlight and prompt for this purpose).

However, your assertion, the prompt interferes with the use of commands through copy and paste, is also reasonable  (especially in all commands are executed in same environment, same directory, same user, and the commands output nothing). So IMO, 

- If we can get some benefit with the prompt, it should be used with `console` highlight.
- If not, it's better to use the prompt and use `bash` highlight.

However, as an exception, it is better to be allowed to use the prompt and `console` highlight in even no benefit areas in order to maintain uniformity throughout the document.

**Comment by EriKWDev** [comment]
I'm not arguing against showing prompts, just the completely unnecessary leading `$` or `>`. 

If a prompt is shown I sure hope it serves to illustrate something and not just be decor.

In your example with docker, the purpose is to show what happened to your environment after the command, but I find that that is the rare exception rather than the more common use case of "here is the command you need to execute"

I doubt anyone reading build instructions are interested in what effect `cd ./build` will have on their current path.

---

I don't have time tonight to fix the issues so if anyone wants to do it just go ahead.

**Pull request merged**